compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java

```diff
@@ -183,9 +183,13 @@ public void visit(BFiniteType bFiniteType) {
 
     @Override
     public void visit(BInvokableType bInvokableType) {
-        buff.writeInt(bInvokableType.paramTypes.size());
-        for (BType params : bInvokableType.paramTypes) {
-            writeTypeCpIndex(params);
+        if (Symbols.isFlagOn(bInvokableType.flags, Flags.ANY_FUNCTION)) {
```

Comment:
Same concern here. But if we are changing this I guess we need to change the BIR too.

Reply:
Fixed.